### DISCUSSION TITLE
Inside a sync scope what order do tables sync?

### DISCUSSION THREAD

**Discussion by jantoineqci** [Mimetis/Dotmim.Sync#1287] [Q&A]
I have a sync scope with 10 tables. I list Table A before Table B and even alphabetically Table A becomes before Table B. Table B has FK to Table A. When I sync I get an error saying The FK value of Table B doesn't exists in Table A. What is the order of tables processed in a sync? or can I dictate the order in which the tables are processed?

**Comment by Mimetis** [comment]
The order is declared when you are creating your Setup
The order should always be **"Parent" - "Child"** 

For instance, as example, your setup for **AdventureWorks** should be something like:

``` cs
var setup = new SyncSetutp("Address", "Customer", "CustomerAddress");
```

Because we need the `Address` rows and `Customer` rows before creating the relations in `CustomerAddress`.
During sync, **DMS** is:
- Doing firstly the `Update` / `Insert` operation in the order of your declared tables.
- Doing secondly `Delete` operations in **reverse** order of your declared tables.

If you still have some foreign keys constraints issues, you can try the options `DisableConstraintsOnApplyChanges`

``` cs
var options = new SyncOptions { DisableConstraintsOnApplyChanges = true };
```

**Comment by HansenBerlin** [comment]
For some reason, regardless of the order of my tables in the setup, provisioning takes place in a different order (as I can see in the progress) and therefore leads to issues due to missing references (even with DisableConstraintsOnApplyChanges = true).

**Reply by Mimetis** [reply]
The order is determined by the tables order you specified:

``` cs
var setup = new SyncSetutp("Address", "Customer", "CustomerAddress")
```

For Insert  / Update, the order is `Address `then `Customer`, then `CustomerAddress`
For Delete, the order is the opposite: `CustomerAddress`, then `Customer `then `Address`
